test/test_mms.py

```diff
@@ -0,0 +1,279 @@
+            reactant2=traps,
+            product=trapped,
+            k_0=k_0,
+            E_k=E_k,
+            p_0=p_0,
+            E_p=E_p,
+            volume=vol,
+        )
+    ]
+
+    my_model.temperature = T_expr
+
+    my_model.boundary_conditions = [
+        F.DirichletBC(subdomain=left, value=mobile_analytical_ufl, species=mobile),
+        F.DirichletBC(subdomain=right, value=mobile_analytical_ufl, species=mobile),
+        F.DirichletBC(subdomain=left, value=trapped_analytical_ufl, species=trapped),
+        F.DirichletBC(subdomain=right, value=trapped_analytical_ufl, species=trapped),
+    ]
+
+    my_model.sources = [
+        F.Source(value=f, volume=vol, species=mobile),
+        F.Source(value=g, volume=vol, species=trapped),
+    ]
+
+    my_model.settings = F.Settings(atol=1e-12, rtol=1e-12, transient=False)
+
+    my_model.initialise()
+    my_model.run()
+
+    mobile_computed = mobile.post_processing_solution
+    trapped_computed = trapped.post_processing_solution
+
+    L2_error_mobile = error_L2(mobile_computed, mobile_analytical_np)
+    L2_error_trapped = error_L2(trapped_computed, trapped_analytical_np)
+
+    assert L2_error_mobile < 2e-07
+    assert L2_error_trapped < 1e-07
+
+
+def test_MMS_transient():
+    """
+    MMS test with one mobile species in 0.1s transient
+    """
+
+    final_time = 0.1
+
+    def u_exact(mod):
+        return lambda x, t: 1 + mod.sin(2 * mod.pi * x[0]) + t**2
+
+    def u_exact_alt(mod):
+        return lambda x: 1 + mod.sin(2 * mod.pi * x[0]) + final_time**2
+
+    H_analytical_ufl = u_exact(ufl)
+    H_analytical_np = u_exact_alt(np)
+
+    elements = ufl.FiniteElement("CG", test_mesh_1d.mesh.ufl_cell(), 1)
+    V = fem.FunctionSpace(test_mesh_1d.mesh, elements)
+    T = fem.Function(V)
+
+    D_0 = 1
+    E_D = 0.1
+    T_expr = lambda x: 600 + 50 * x[0]
+    T.interpolate(T_expr)
+    D = D_0 * ufl.exp(-E_D / (F.k_B * T))
+
+    # FESTIM model
+
+    my_model = F.HydrogenTransportProblem()
+    my_model.mesh = test_mesh_1d
+    my_mat = F.Material(name="mat", D_0=D_0, E_D=E_D)
+    vol = F.VolumeSubdomain1D(id=1, borders=[0, 1], material=my_mat)
+    left = F.SurfaceSubdomain1D(id=1, x=0)
+    right = F.SurfaceSubdomain1D(id=2, x=1)
+
+    my_model.subdomains = [vol, left, right]
+
+    H = F.Species("H")
+    my_model.species = [H]
+
+    my_model.temperature = T_expr
+
+    my_model.boundary_conditions = [
+        F.DirichletBC(subdomain=left, value=H_analytical_ufl, species=H),
+        F.DirichletBC(subdomain=right, value=H_analytical_ufl, species=H),
+    ]
+
+    init_value = lambda x: 1 + ufl.sin(2 * ufl.pi * x[0])
+    my_model.initial_conditions = [F.InitialCondition(value=init_value, species=H)]
+
+    my_model.settings = F.Settings(atol=1e-10, rtol=1e-10, final_time=final_time)
+    my_model.settings.stepsize = final_time / 20
+
+    my_model.initialise()
+
+    f = lambda x, t: (
+        (H_analytical_ufl(x, t) - H.prev_solution) / my_model.dt
```

Comment:
No you shouldn't have this in MMS.
If your manufactured solution is:

$$c =  1 + \sin{2 \pi \ x} + t^2$$

then

$$ \frac{\partial c}{\partial t} = 2 \ t $$


The form c - c_n / delta_t is not an exact solution, it's an approximation of dc/dt !!